internal/sonar/sonar.go

```diff
@@ -0,0 +1,81 @@
+package sonar
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type SonarConfig struct {
+	ProjectKey string
+	APIToken   string
+}
+
+type SonarResults struct {
+	Component Component `json:"component"`
+}
+
+type Component struct {
+	Id        string     `json:"id,omitempty"`
+	Key       string     `json:"key"`
+	Name      string     `json:"name"`
+	Qualifier string     `json:"qualifier"`
+	Measures  []Measures `json:"measures"`
+}
+
+type Measures struct {
+	Metric string `json:"metric"`
+	Value  string `json:"value"`
+}
+
+func NewSonarConfig(projectKey, apiToken string) *SonarConfig {
+	return &SonarConfig{
+		ProjectKey: projectKey,
+		APIToken:   apiToken,
+	}
+}
+
+func (sc *SonarConfig) GetSonarResults() (*SonarResults, error) {
+	httpClient := &http.Client{}
+	var url string
+	var token string
+
+	if sc.ProjectKey != "" && sc.APIToken != "" {
+		/* SonarQube works differently to SonarCloud - with Qube, each company has their own specific
```

Comment:
is this comment still relevant?

Reply:
I left it in so that if we decide to expand the command to include SonarQube in the future we have a reminder of what the differences are/possible solutions. Can delete if we'd rather have tidier code.

Reply:
no worries, we can keep it